plenum/test/node_catchup/test_config_ledger.py

```diff
@@ -0,0 +1,141 @@
+import json
+
+import pytest
+
+from plenum.test.node_catchup.helper import waitNodeDataInequality, \
+    waitNodeDataEquality
+from plenum.test.pool_transactions.helper import \
+    disconnect_node_and_ensure_disconnected, reconnect_node_and_ensure_connected
+
+from plenum.common.util import randomString
+from plenum.test.helper import send_signed_requests, sign_requests, \
+    waitForSufficientRepliesForRequests, check_sufficient_replies_received
+
+from plenum.common.constants import CONFIG_LEDGER_ID, DATA
+from plenum.test.test_config_req_handler import write_conf_op, \
+    TestConfigReqHandler, WRITE_CONF, READ_CONF, read_conf_op
+from plenum.test.test_node import TestNode
+from stp_core.loop.eventually import eventually
+
+
+class NewTestNode(TestNode):
+    def getConfigReqHandler(self):
+        return TestConfigReqHandler(self.configLedger,
+                                self.states[CONFIG_LEDGER_ID])
+
+
+def write(key, val, looper, client, wallet):
+    reqs = send_signed_requests(client,
+                                sign_requests(wallet,
+                                              [write_conf_op(key, val)]))
+    waitForSufficientRepliesForRequests(looper, client,
+                                        requests=reqs)
+
+
+def read(key, looper, client, wallet):
+    req, = send_signed_requests(client, sign_requests(wallet,
+                                                      [read_conf_op(key)]))
+    resp = looper.run(eventually(check_sufficient_replies_received, client,
+                                 req.identifier, req.reqId, retryWait=1))
+    return json.loads(resp[DATA])[key]
+
+
+def send_some_config_txns(looper, client, wallet, keys):
+    for i in range(5):
+        key, val = 'key_{}'.format(i+1), randomString()
+        write(key, val, looper, client, wallet)
+        keys[key] = val
+    return keys
+
+
+@pytest.fixture(scope="module")
+def testNodeClass(patchPluginManager):
+    return NewTestNode
+
+
+@pytest.fixture(scope="module")
+def setup(testNodeClass, txnPoolNodeSet):
+    for node in txnPoolNodeSet:
+        ca = node.clientAuthNr.core_authenticator
+        ca.write_types.add(WRITE_CONF)
```

Comment:
Do we need to call this for all new ledgers? Can we make a general support of this on node level?

Reply:
No, only whose request authentication is done using domain ledger, a practical implementation would be to use core_authenticator and some roles check as done in one of the plugin

Reply:
But how do real plugins can add support for new txn types for existing ledgers?
And I think that if we have a better way of registering plugins, then we don't have the code like above (see my comment about real plugins registering)

Reply:
They define a authenticator class and register that authenticator in `update_node_obj`, i will push an example of that